### PR TITLE
GeoJSON revamp: data in repo

### LINKED ISSUE #109: Mapping: Ignore 'Units' and disaggregation columns
Having a 'Units' column in the CSV prevents the map from displaying data. Can we hide the 'Units' column from the mapping script as a short-term solution, and display the first Units option by default?

Alternatively, we could include a 'Units' option in the map, but this implies to users that the disaggregations are available to be used on the map, which they aren't (yet).

**Note**: This issue is a problem with disaggregation in general, as the presence of Geographical data overlapping with another category in the CSV prevents data from displaying on the map. We would like to have disaggregations working on the map in future, but this is likely to be complex.

### PR DESCRIPTION
**Pull request by brockfanning**
Fixes #109 

This depends on https://github.com/open-sdg/sdg-build/pull/99 so a full review needs to be done of that PR as well as this one.

## Breaking changes

This changes the map functionality so that it expects the GeoJSON to be in the data repository. This needs to happen in the data repository via sdg-build. So, if a country upgrades to this version of open-sdg without also upgrading sdg-build versions, their maps will stop functioning.

Along the same lines, the map configuration in _config.yml needs to change.

## Overview of changes

1. The map no longer looks for a single GeoJSON file at a hardcoded location. Instead, it looks for individual GeoJSON files in the data repo that have been pre-built for each indicator.
2. The GeoJSON download button on the maps has been removed, and standard links have been added below the map. Some benefits of this:
    a. They use the same style as other download buttons in the platform
    b. The download is a simple link, so it can be reached by scripts (interoperability)
    c. Instead of confusingly only downloading the visible layer, it now displays all the layers as separate buttons
3. A lot of Javascript code in open-sdg has been removed because it isn't needed
4. Several javascript libraries are no longer being loaded on non-map pages (saves bandwidth)
5. I've removed any reference to the `data_geocode_regex` metadata field. I think this was non-functional anyway, and we would need to reimplement that functionality if it's still wanted.
6. The GeoJSON files now contains all the disaggregation data, though it's currently just using the first one. (After this is merged I'll post a followup PR that displays all the disaggregation, fixing #374.)
7. The options in the map configuration have changed.

## Screenshots

Here is an example of the new download button (the "My English regions" is configurable and just demonstrates that it is translated):
Screenshot from 2020-02-11 09-52-01

### CODE REVIEW

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin @SavvasStephanides A lot to go over with this one. We could discuss in the next call. Testing is a bit of a hassle because it depends on an sdg-build change.

**Comment by christoph4711** [comment]
Where I can test the new map functionalities?
And how can I update my starter repo? (Sorry for this question, but I am not familiar with this github stuff)

**Comment by brockfanning** [comment]
@christoph4711 Can you check what version of open-sdg you're using now? It would be in the `remote_theme` line of `_config.yml`. Also can you check the version of jekyll-open-sdg-plugins that you're using (this would be in `Gemfile`). Finally, in the data repository, what version of sdg-build are you using? (this would be in the `scripts/requirements.txt` file in the data repository).

The reason I ask is that in order to test cutting-edge features, you'll need to be using the latest versions of all those things.

Also for your awareness, we are on track to release a stable version 1.0.0 of Open SDG some time in the next month. When we release that, we'll include thorough upgrade instructions. So it may be more efficient to hold off.

In the meantime, I'll try to add some screenshots to this PR so you can see any visual changes.

**Comment by brockfanning** [comment]
One nice thing about this - this makes it almost trivial to add a disaggregation drop-down to the maps. In fact, I'm almost tempted to add it to this PR... Here's an example of what it might look like:
Screenshot from 2020-02-11 09-48-51

**Comment by christoph4711** [comment]
@brockfanning 
This feature would be great! I try to hold off  until 1.0.0. :-)

**Comment by brockfanning** [comment]
Since it was so easy to add, I went ahead and added the disaggregation select to this PR. I've updated the PR description and title.

**Comment by brockfanning** [comment]
@AnnCorp @SavvasStephanides @LucyGwilliamAdmin I've removed the disaggregation select from this PR, and updated the description/title. I can make a follow-up PR after this for continuing with that feature.

Any review/testing is welcome! As a reminder, it depends on another PR in sdg-build.

**Comment by brockfanning** [comment]
@SavvasStephanides @LucyGwilliamAdmin This should be a bit easier to test, now that the sdg-build change has been merged.

**Comment by brockfanning** [comment]
@SavvasStephanides @LucyGwilliamAdmin Just a note that I resolved the merge conflicts on this one.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @SavvasStephanides really sorry so I think I've missed something about the make serve stuff?

How do I test this using make serve?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin In theory, if you have the required software installed (Ruby, Python, Node.js) you can test it by pull my branch and running `make serve` in the root of the repository. If you don't already have my remote set up in Git, the command would be:
```
git remote add brock https://github.com/brockfanning/open-sdg.git
```
Then you could do this:
```
git fetch brock
git checkout geojson-in-data-repo
make serve
```

**Comment by LucyGwilliamAdmin** [comment]
Thanks @brockfanning 

So I've got all the software and this worked right up until the last line (`make serve`), where I got the error:
'make' is not recognized as an internal or external command

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin Ah, I think that it doesn't work in Windows. I guess this is a good reason we need the Docker solution @SavvasStephanides is working on.

**Comment by LucyGwilliamAdmin** [comment]
Ok no worries, thanks for you help though! I think I'll have to leave this one all to you @SavvasStephanides

**Comment by SavvasStephanides** [comment]
@LucyGwilliamAdmin If you can run Docker you can give it a go https://github.com/open-sdg/open-sdg/issues/197#issuecomment-594548506

**Comment by SavvasStephanides** [comment]
@brockfanning Do you have an indicator I can test this on?

**Comment by brockfanning** [comment]
@SavvasStephanides The tests for this use indicator 1.3.1.

**Comment by SavvasStephanides** [comment]
Thanks @brockfanning ! Seems to be working nicely. I'll have a quick check on the code and approve.

File: _includes/assets/js/plugins/jquery.sdgMap.js [hunk 1]
```diff
@@ -2,7 +2,11 @@
  * TODO:
  * Integrate with high-contrast switcher.
  */
-(function($, L, chroma, window, document, undefined) {
+(function($) {
+
+  if (typeof L === 'undefined') {
```
Comment (SavvasStephanides) [inline _includes/assets/js/plugins/jquery.sdgMap.js]: Where does `L` point to here, if you removed it from the function's arguments?

Comment (brockfanning) [reply _includes/assets/js/plugins/jquery.sdgMap.js]: @SavvasStephanides That's a global variable created by Leaflet. It's not there, I assume that Leaflet did not get successfully loaded.

Comment (SavvasStephanides) [reply _includes/assets/js/plugins/jquery.sdgMap.js]: Ah fair enough. Thanks!

**Pull request merged**